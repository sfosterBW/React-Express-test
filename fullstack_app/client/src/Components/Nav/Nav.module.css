.wrapper {
  width: 100%;
  display: flex;
}

.left, .right {
  display: flex;
  box-sizing: border-box;
  align-items: center;
  padding: 1vmin;
}

.right {
  margin-left: auto;
  margin-right: 2vmin;
}

.logo, .title, .link {
  display: inline-block;
  box-sizing: border-box;
  vertical-align: middle;
  margin: 0 2vmin;
}

.logo {

}

.title {
  font-size: calc(10px + 2vmin);
  font-weight: bold;
}

.link {
  font-size: calc(4px + 2vmin);
}
