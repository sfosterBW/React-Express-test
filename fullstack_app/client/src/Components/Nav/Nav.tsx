import React, { FC } from 'react'
import { Link } from '@reach/router'
import styles from './Nav.module.css'

const Nav: FC = () => {

  return (
    <header>
      <nav className={styles.wrapper}>
        <div className={styles.left}>
          <Link to="/">
          <img
            alt="logo"
            className={styles.logo}
            src="https://img.icons8.com/cotton/64/000000/apple--v1.png" />
          </Link>
          <h2 className={styles.title}>Fruit Dashboard</h2>
        </div>
        <div className={styles.right}>
          <Link
            className={styles.link}
            to="add"
            rel="nofollower noreferrer"
            title="Add a new fruit to your list">
            Add new fruit
          </Link>
          <Link
            className={styles.link}
            to="edit"
            rel="nofollower noreferrer"
            title="View and edit your list of fruit">
            Edit fruit
          </Link>
        </div>
      </nav>
    </header>
  )
}

export default Nav
