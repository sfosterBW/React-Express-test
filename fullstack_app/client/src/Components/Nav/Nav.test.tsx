import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Nav from './Nav'

describe('the Nav component', () => {

  const component = <Nav />
  const wrapper = mount(component)

  it('renders with the correct structure', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper.find('img')).toHaveLength(1)
    expect(wrapper.find('h2')).toHaveLength(1)
    expect(wrapper.find('Link')).toHaveLength(3)
  })

  it('functions as expected',() => {
    expect(wrapper.find('img').props().alt).toBe('logo')
    expect(wrapper.find('h2').text()).toBe('Fruit Dashboard')
    expect(wrapper.find('Link').at(0).props().to).toBe("/")
    expect(wrapper.find('Link').at(1).props().to).toBe("add")
    expect(wrapper.find('Link').at(2).props().to).toBe("edit")
  })

  it('renders the same as last time', () => {
    const tree = renderer
      .create(component)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
