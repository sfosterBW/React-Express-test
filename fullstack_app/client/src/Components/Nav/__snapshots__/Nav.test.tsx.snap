// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Nav component renders the same as last time 1`] = `
<header>
  <nav
    className="wrapper"
  >
    <div
      className="left"
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="logo"
          src="https://img.icons8.com/cotton/64/000000/apple--v1.png"
        />
      </a>
      <h2
        className="title"
      >
        Fruit Dashboard
      </h2>
    </div>
    <div
      className="right"
    >
      <a
        className="link"
        href="/add"
        onClick={[Function]}
        rel="nofollower noreferrer"
        title="Add a new fruit to your list"
      >
        Add new fruit
      </a>
      <a
        className="link"
        href="/edit"
        onClick={[Function]}
        rel="nofollower noreferrer"
        title="View and edit your list of fruit"
      >
        Edit fruit
      </a>
    </div>
  </nav>
</header>
`;
